sysl2/sysl/ints.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+
+	"github.com/anz-bank/sysl/src/proto"
+	log "github.com/sirupsen/logrus"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+type IntegrationParam struct {
+	m            *sysl.Module
+	excludes     StrSet
+	passthroughs StrSet
+	commonEndpts StrSet
+	highlights   StrSet
+	apps         StrSet
+	deps         map[string]*AppDependency
+}
+
+func MakeIntegrationParam(mod *sysl.Module, excludes, passthroughs, highlights StrSet) *IntegrationParam {
+	return &IntegrationParam{
+		m:            mod,
+		excludes:     excludes,
+		passthroughs: passthroughs,
+		commonEndpts: MakeStrSet(".. * <- *", "*"),
+		highlights:   highlights,
+		apps:         StrSet{},
+		deps:         map[string]*AppDependency{},
+	}
+}
+
+func (v *IntegrationParam) collectApplicationDependencies(apps StrSet) {
+	for app := range apps {
+		for epname, endpt := range v.m.GetApps()[app].GetEndpoints() {
```

Comment:
`apps := v.m.GetApps()` and use `apps` in the loop.

Reply:
changed